_posts/2023-03-15-shuffling-large-data-at-constant-memory.md

```diff
@@ -0,0 +1,212 @@
+
+The biggest benefit of P2P shuffling is constant memory usage.
+Memory usage drops and stays constant across all workloads:
+
+<figure class="align-center">
+<img alt="P2P shuffling uses significantly less memory than task-based shuffling on all workloads." src="/images/p2p_benchmark_suite.png" style="width: 100%;"/>
+<figcaption>
+<em>P2P shuffling (blue) uses constant memory while task-based shuffling (orange) scales linearly.</em>
+</figcaption>
+</figure>
+
+For the tested workloads, we saw up to **10x lower memory**.
+For even larger workloads, this gap only increases.
+
+### Performance and Speed
+
+In the above plot, we can two performance improvements:
+
+- **Faster execution:** Workloads run up to 45% faster.
+- **Quicker startup:** Smaller graphs mean P2P shuffling starts sooner.
+
+Comparing scheduler dashboards for P2P and task-based shuffling side-by-side
+helps to understand what causes these performance gains:
+
+<figure class="align-center">
+<img alt="Task-based shuffling shows bigger gaps in the task stream and spilling compared to P2P shuffling." src="/images/dashboard-comparison.png" style="width: 100%;"/>
+<figcaption>
+<em>Task-based shuffling (left) shows bigger gaps in the task stream and spilling compared to P2P shuffling (right).</em>
+</figcaption>
+</figure>
+
+- **Graph size:** 10x fewer tasks are faster to deploy and easier on the scheduler.
+- **Controlled I/O:** P2P shuffling handles I/O explicitly, which avoids less performant spilling by Dask.
+- **Fewer interruptions:** I/O is well balanced, so we see fewer gaps in work in the task stream.
+
+Overall, P2P shuffling leverages our hardware far more effectively.
+
+## Changing defaults
+
+These results benefit most users, so P2P shuffling is now the default starting from release `2023.2.1`,
+as long as `pyarrow>=7.0.0` is installed. Dataframe operations like the following will benefit:
+
+- `df.set_index(...)`
+- `df.merge(...)`
+- `df.groupby(...).apply(...)`
+
+### Share your experience
+
+To understand how P2P performs on the vast range of workloads out there in the wild, and to ensure that making it the default was the right choice, we would love to learn about your experiences running it.
+We have opened a [discussion issue](https://github.com/dask/distributed/discussions/7509) on GitHub for feedback on this change.
+Please let us know how it helps (or doesn’t).
+
+### Keep old behavior
+
+If P2P shuffling does not work for you, you can deactivate it by setting the `dataframe.shuffle.method` config value to `"tasks"` or explicitly setting a keyword argument, for example:
+
+with the `yaml` config
+
+```yaml
+dataframe:
+  shuffle:
+    method: tasks
+```
+
+or when starting a Coiled cluster
+
+```python3
+import dask
+import coiled
+
+# The dataframe.shuffle.method config is available since 2023.3.1
+with dask.config.set({"dataframe.shuffle.method": "tasks"}):
+    cluster = coiled.Cluster(...)  # coiled sends current dask config automatically
+```
```

Comment:
Could we change this to use an OSS cluster manager? 

````suggestion
or when using a cluster manager

```python3
import dask
from dask.distributed import LocalCluster

# The dataframe.shuffle.method config is available since 2023.3.1
with dask.config.set({"dataframe.shuffle.method": "tasks"}):
    cluster = LocalCluster(...)  # many cluster managers send current dask config automatically
```
````